Editor/Modules/ShaderAnalyzer.cs

```diff
@@ -0,0 +1,59 @@
+using System.Collections.Generic;
+using Unity.ProjectAuditor.Editor.Core;
+using Unity.ProjectAuditor.Editor.Diagnostic;
+using Unity.ProjectAuditor.Editor.Utils;
+using UnityEngine;
+using UnityEngine.Rendering;
+
+namespace Unity.ProjectAuditor.Editor.Modules
+{
+    class ShaderAnalyzer : IShaderModuleAnalyzer
+    {
+
+        internal const string PAS0000 = nameof(PAS0000);
+
+        internal static readonly Descriptor k_SrpBatcherDescriptor = new Descriptor(
+            PAS0000,
+            "Shader: Not compatible with SRP batcher",
+            Area.GPU,
+            "The shader is not compatible with SRP batcher.",
+            "Consider fixing the shader, if the SRP batcher compatibility was not intentionally removed."
```

Comment:
I would rephrase as "Consider adding SRP batcher compatibility to the shader. This will reduce the CPU time Unity requires to prepare and dispatch draw calls for materials that use the same shader variant."

Reply:
Done